File: Backend/Clinica.Dominio/Entidades/Profesional.cs
namespace Clinica.Dominio.Entidades;

public class Profesional
{
    public int Id { get; set; }
    public string Nombre { get; set; } = string.Empty;
    public string Apellido { get; set; } = string.Empty;
    public string Especialidad { get; set; } = string.Empty;

    // Cuánto dura cada turno de este profesional: define la grilla de
    // horarios que se ofrece al pedir/reprogramar un turno (ver
    // ServicioTurnos.ObtenerHorariosDisponiblesAsync). Lo puede modificar el
    // propio profesional o el administrador.
    public int DuracionTurnoMinutos { get; set; } = 30;

    // Email de contacto (distinto del NombreUsuario con el que loguea el
    // profesional en su Usuario vinculado). Opcional: no se pide al crear.
    public string? Email { get; set; }

    // Un profesional desactivado no puede loguearse (se desactiva también su
    // Usuario vinculado) ni se lo ofrece para pedir turnos nuevos, pero
    // conserva su historial de turnos. Ver ServicioProfesionales.
    public bool Activo { get; set; } = true;
    public DateTime? FechaDesactivacion { get; set; }

    public DateTime FechaCreacion { get; set; }

    public ICollection<Turno> Turnos { get; set; } = new List<Turno>();

    // Días y horarios en los que atiende. Vacío significa "todavía no
    // configuró un horario propio": ServicioTurnos cae en ese caso al
    // horario general de la clínica (ver HorarioClinica), para no romper a
    // los profesionales existentes que no lo hayan cargado.
    public ICollection<BloqueHorarioProfesional> BloquesHorario { get; set; } = new List<BloqueHorarioProfesional>();
}

File: Frontend/src/modelos/profesional.ts
// Coincide con el enum DayOfWeek de .NET: el backend lo serializa como texto
// en inglés (JsonStringEnumConverter), empezando por domingo.
export type DiaSemana = 'Sunday' | 'Monday' | 'Tuesday' | 'Wednesday' | 'Thursday' | 'Friday' | 'Saturday';

export const DIAS_SEMANA: { valor: DiaSemana; etiqueta: string }[] = [
  { valor: 'Monday', etiqueta: 'Lunes' },
  { valor: 'Tuesday', etiqueta: 'Martes' },
  { valor: 'Wednesday', etiqueta: 'Miércoles' },
  { valor: 'Thursday', etiqueta: 'Jueves' },
  { valor: 'Friday', etiqueta: 'Viernes' },
  { valor: 'Saturday', etiqueta: 'Sábado' },
  { valor: 'Sunday', etiqueta: 'Domingo' },
];

export interface BloqueHorario {
  diaSemana: DiaSemana;
  // "HH:mm:ss", tal cual serializa TimeOnly en el backend.
  horaInicio: string;
  horaFin: string;
}

export interface Profesional {
  id: number;
  nombre: string;
  apellido: string;
  especialidad: string;
  duracionTurnoMinutos: number;
  email: string | null;
  activo: boolean;
  // Calculado por el backend: desactivado hace 7+ días y sin turnos
  // asociados. Controla si se puede mostrar el botón "Eliminar".
  puedeEliminarse: boolean;
  // Días y horarios en los que atiende. Vacío significa "sin horario propio
  // configurado" (el backend cae en ese caso al horario general de la
  // clínica, ver ServicioTurnos en el backend).
  horarios: BloqueHorario[];
}

export interface GuardarProfesionalDto {
  nombre: string;
  apellido: string;
  especialidad: string;
  duracionTurnoMinutos: number;
  // Opcional: en la edición, dejarlo vacío no cambia el email actual (se
  // manda tal cual, incluso vacío, a diferencia de nuevaContrasena).
  email: string;
  // Opcional: dejarlo vacío en la edición no cambia la contraseña actual.
  nuevaContrasena?: string;
}

// Solo se usa al dar de alta: el administrador define acá el usuario y
// contraseña con los que ese profesional va a loguearse (ver
// ServicioProfesionales.CrearAsync en el backend, que crea el Usuario
// vinculado en el mismo paso).
export interface CrearProfesionalDto extends GuardarProfesionalDto {
  nombreUsuario: string;
  contrasena: string;
}
